pkg/schemacompat/schemacompat_test.go

```diff
@@ -0,0 +1,217 @@
+package schemacompat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestSuccessNewHasMoreProperties(t *testing.T) {
+	existing := &apiextensionsv1.JSONSchemaProps{
+		Type: "object",
+		Properties: map[string]apiextensionsv1.JSONSchemaProps{
+			"existing": {
```

Comment:
style nit: it can sometimes help readability to compactify these struct declarations into a single line, e.g.:

```
"existing": {Type: "string"},
"new":      {Type: "integer"},
```

Reply:
Sure, I'll fix it.